internal/worker/dbaccessor/tracker.go

```diff
@@ -179,7 +180,10 @@ func (w *trackedDBWorker) ensureModelDBInitialised(ctx context.Context) error {
 // should use.
 func (w *trackedDBWorker) Txn(ctx context.Context, fn func(context.Context, *sqlair.TX) error) error {
 	return w.run(ctx, func(db *sqlair.DB) error {
-		return errors.Trace(database.Txn(w.tomb.Context(ctx), db, fn))
+		// Tie the worker tomb to the context, so that the worker can be
```

Comment:
This comment is not quite right. It would seem to indicate (at least to me) that cancelling the context kills the tomb, but that is not the behaviour of `w.tomb.Context(ctx)`.

I would explicitly say that it is so that we can choose whether to kill the worker based on the context's error.